src/experimental/coroutines.rs

```diff
@@ -91,8 +168,13 @@ impl Future for TimerDelayFuture {
 }
 
 pub fn wait_seconds(time: f32) -> TimerDelayFuture {
+    let now = get_context()
+        .coroutines_context
+        .active_coroutine_now
+        .unwrap_or_else(|| miniquad::date::now());
+
     TimerDelayFuture {
-        start_time: miniquad::date::now(),
+        start_time: now,
```

Comment:
Alternatively you could remove `start_time` and replace it with `remaining_time`. Then, in each tick, you do `self.remaining_time -= get_frame_time()` and check if it is negative to decide whether to finish the future or keep pending

Reply:
Thanks, fixed, `remaining_time` is better!